### PR TITLE
Fix examples

### PR DESCRIPTION
**Pull request by H4ckerxx44**
## Summary
"Modernize" examples. 
Previous Pull-Request (#33)

- Unified example class and instance names
- Use nextcord.ext instead of on_message for command handling
- Set prefix to the same in every example

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -70,44 +70,32 @@ Please note that on Linux installing voice you must install the following packag
...
+    class Bot(commands.Bot):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+        async def on_ready(self):
+            print(f'Logged in as {self.user} (ID: {self.user.id})')
+            print('------')
+
+
+    bot = Bot(command_prefix=commands.when_mentioned_or('$'))
```
Comment (TAG-Epic) [inline README.rst:90]: Having mentioned_or is more advanced usage.

File: README.rst [hunk 2]
```diff
@@ -70,44 +70,32 @@ Please note that on Linux installing voice you must install the following packag
...
-            if message.author == self.user:
-                return
-
-            if message.content == 'ping':
-                await message.channel.send('pong')
-
-    client = MyClient()
-    client.run('token')
 
 Bot Example
```
Comment (TAG-Epic) [inline README.rst:74]: Maybe rename this to Quick Example

File: README.rst [hunk 3]
```diff
@@ -70,44 +70,32 @@ Please note that on Linux installing voice you must install the following packag
...
+            print('------')
+
+
+    bot = Bot(command_prefix=commands.when_mentioned_or('$'))
 
     @bot.command()
-    async def ping(ctx):
-        await ctx.send('pong')
+    async def hello(ctx: commands.Context):
+        await ctx.reply('Hello!')
```
Comment (TAG-Epic) [inline README.rst:94]: .send should be used here

File: README.rst [hunk 4]
```diff
@@ -70,44 +70,32 @@ Please note that on Linux installing voice you must install the following packag
...
+            print(f'Logged in as {self.user} (ID: {self.user.id})')
+            print('------')
+
+
+    bot = Bot(command_prefix=commands.when_mentioned_or('$'))
 
     @bot.command()
-    async def ping(ctx):
-        await ctx.send('pong')
+    async def hello(ctx: commands.Context):
```
Comment (TAG-Epic) [inline README.rst:93]: keep it as ping and it doing the same thing.

Comment (TAG-Epic) [reply README.rst:93]: Typehinting is also a bit too advanced for a first example

File: README.rst [hunk 5]
```diff
@@ -70,44 +70,32 @@ Please note that on Linux installing voice you must install the following packag
...
 Bot Example
 ~~~~~~~~~~~~~
 
 .. code:: py
 
-    import nextcord
     from nextcord.ext import commands
 
-    bot = commands.Bot(command_prefix='>')
+    class Bot(commands.Bot):
```
Comment (TAG-Epic) [inline README.rst:81]: Subclassing it is unneccesary

File: docs/intro.rst [hunk 6]
```diff
@@ -99,15 +99,17 @@ A quick example to showcase how events work:
 
 .. code-block:: python3
 
-    import nextcord
+    from nextcord.ext import commands
 
-    class MyClient(nextcord.Client):
-        async def on_ready(self):
-            print(f'Logged on as {self.user}!')
+    class Bot(commands.Bot):
```
Comment (TAG-Epic) [inline docs/intro.rst:104]: Unnecessary subclass

File: docs/intro.rst [hunk 7]
```diff
@@ -99,15 +99,17 @@ A quick example to showcase how events work:
...
+            super().__init__(*args, **kwargs)
 
-        async def on_message(self, message):
+        async def on_message(self, message: nextcord.Message):
             print(f'Message from {messsage.author}: {message.content}')
 
-    client = MyClient()
-    client.run('my token goes here')
+
+    bot = Bot(command_prefix=commands.when_mentioned_or('$'))
```
Comment (TAG-Epic) [inline docs/intro.rst:112]: maybe leave a comment saying the prefix is not used in this example

File: examples/basic_bot.py [hunk 8]
```diff
@@ -12,20 +12,20 @@
...
+bot = commands.Bot(command_prefix=commands.when_mentioned_or('$'), description=description, intents=intents)
 
 @bot.event
 async def on_ready():
     print(f'Logged in as {bot.user} (ID: {bot.user.id})')
     print('------')
 
 @bot.command()
-async def add(ctx, left: int, right: int):
+async def add(ctx: commands.Context, left: int, right: int):
```
Comment (TAG-Epic) [inline examples/basic_bot.py:23]: No need to typehint context here

File: examples/edits.py [hunk 9]
```diff
@@ -1,20 +1,34 @@
...
 
-class MyClient(nextcord.Client):
+from nextcord.ext import commands
+
+
+class Bot(commands.Bot):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
     async def on_ready(self):
```
Comment (TAG-Epic) [inline examples/edits.py:11]: on_ready should be removed from all examples where that isn't the example like basic_bot or the one in the readme

**Review by TAG-Epic — CHANGES_REQUESTED** [review]
Else it looks pretty good to me

File: README.rst (hunk 3 above)
Comment (H4ckerxx44) [reply README.rst:94]: I don't think so. The original was [this](https://github.com/Rapptz/discord.py/blob/master/examples/reply.py) and it specifically replied to the user.

Comment (TAG-Epic) [reply README.rst:94]: My bad.

File: README.rst (hunk 1 above)
Comment (H4ckerxx44) [reply README.rst:90]: Should I keep it for the actual examples? I would choose to.

Comment (TAG-Epic) [reply README.rst:90]: hmmm no

**Comment by H4ckerxx44** [comment]
Changed the mentioned things, tell me if I missed something, if this is the case, I say sorry in before hand.

File: docs/intro.rst [hunk 10]
```diff
@@ -99,15 +99,15 @@ A quick example to showcase how events work:
...
 .. code-block:: python3
 
-    import nextcord
+    from nextcord.ext import commands
 
-    class MyClient(nextcord.Client):
-        async def on_ready(self):
-            print(f'Logged on as {self.user}!')
+    # the prefix is not used in this example
+    bot = commands.Bot(command_prefix=commands.when_mentioned_or('$'))
```
Comment (TAG-Epic) [inline docs/intro.rst:105]: This should not include mentions.

File: examples/basic_bot.py [hunk 11]
```diff
@@ -25,7 +25,7 @@ async def add(ctx, left: int, right: int):
     await ctx.send(left + right)
 
 @bot.command()
-async def roll(ctx, dice: str):
+async def roll(ctx: commands.Context, dice: str):
```
Comment (TAG-Epic) [inline examples/basic_bot.py:28]: Context should not be typehinted here

File: examples/custom_context.py [hunk 12]
```diff
@@ -42,10 +46,5 @@ async def guess(ctx, number: int):
     # or a red cross mark if it wasn't
     await ctx.tick(number == value)
 
-# IMPORTANT: You shouldn't hard code your token
-# these are very important, and leaking them can 
-# let people do very malicious things with your
-# bot. Try to use a file or something to keep
-# them private, and don't commit it to GitHub
 token = "your token here"
 bot.run(token)
```
Comment (TAG-Epic) [inline examples/custom_context.py:50]: These 2 lines could be merged into `bot.run("token")`

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

File: examples/basic_bot.py (hunk 11 above)
Comment (H4ckerxx44) [reply examples/basic_bot.py:28]: It is typehinted everywhere else, remove it in every command then?

Comment (TAG-Epic) [reply examples/basic_bot.py:28]: Yeah

**Comment by H4ckerxx44** [comment]
This should be it now.

File: examples/edits.py [hunk 13]
```diff
@@ -1,20 +1,30 @@
...
+        super().__init__(*args, **kwargs)
+
+    async def on_message(self, message: nextcord.Message):
         if message.content.startswith('!editme'):
             msg = await message.channel.send('10')
             await asyncio.sleep(3.0)
             await msg.edit(content='40')
 
-    async def on_message_edit(self, before, after):
+    async def on_message_edit(self, before: nextcord.Message, after: nextcord.Message):
```
Comment (TAG-Epic) [inline examples/edits.py:17]: This should not be typehinted.

File: examples/new_member.py [hunk 14]
```diff
@@ -1,13 +1,13 @@
...
-        print(f'Logged in as {self.user} (ID: {self.user.id})')
-        print('------')
+from nextcord.ext import commands
+
+class Bot(commands.Bot):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
-    async def on_member_join(self, member):
+    async def on_member_join(self, member: nextcord.Member):
```
Comment (TAG-Epic) [inline examples/new_member.py:10]: This should not be type hinted

File: examples/secret.py [hunk 15]
```diff
@@ -44,7 +44,7 @@ def create_overwrites(ctx, *objects):
 # it is best to lock it to be guild-only.
 @secret.command()
 @commands.guild_only()
-async def text(ctx: commands.Context, name: str, *objects: typing.Union[nextcord.Role, nextcord.Member]):
+async def text(ctx, name: str, *objects: typing.Union[nextcord.Role, nextcord.Member]):
```
Comment (TAG-Epic) [inline examples/secret.py:47]: Name should not be type hinted

File: examples/secret.py [hunk 16]
```diff
@@ -60,7 +60,7 @@ async def text(ctx: commands.Context, name: str, *objects: typing.Union[nextcord
 
 @secret.command()
 @commands.guild_only()
-async def voice(ctx: commands.Context, name: str, *objects: typing.Union[nextcord.Role, nextcord.Member]):
+async def voice(ctx, name: str, *objects: typing.Union[nextcord.Role, nextcord.Member]):
```
Comment (TAG-Epic) [inline examples/secret.py:63]: Name should not be type hinted

File: examples/views/counter.py [hunk 17]
```diff
@@ -1,15 +1,11 @@
...
-class CounterBot(commands.Bot):
-    def __init__(self):
-        super().__init__(command_prefix=commands.when_mentioned_or('$'))
 
-    async def on_ready(self):
-        print(f'Logged in as {self.user} (ID: {self.user.id})')
-        print('------')
+class Bot(commands.Bot):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```
Comment (TAG-Epic) [inline examples/views/counter.py:8]: Do you really need a custom bot class here

File: examples/views/link.py [hunk 18]
```diff
@@ -27,11 +23,11 @@ def __init__(self, query: str):
         self.add_item(nextcord.ui.Button(label='Click Here', url=url))
 
 
-bot = GoogleBot()
+bot = Bot(command_prefix='$')
 
 
 @bot.command()
-async def google(ctx: commands.Context, *, query: str):
+async def google(ctx, *, query: str):
```
Comment (TAG-Epic) [inline examples/views/link.py:30]: Query should not be type hinted here

File: examples/views/persistent.py [hunk 19]
```diff
@@ -27,9 +28,9 @@ async def grey(self, button: nextcord.ui.Button, interaction: nextcord.Interacti
         await interaction.response.send_message('This is grey.', ephemeral=True)
 
 
-class PersistentViewBot(commands.Bot):
-    def __init__(self):
-        super().__init__(command_prefix=commands.when_mentioned_or('$'))
+class Bot(commands.Bot):
+    def __init__(self, *args, **kwargs):
```
Comment (TAG-Epic) [inline examples/views/persistent.py:32]: Remove every custom bot class that isn't required

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

**Review by TAG-Epic — APPROVED** [review]
LGTM

**Pull request merged**